keras_nlp/models/whisper/whisper_preprocessor_test.py

```diff
@@ -0,0 +1,220 @@
+            x["encoder_features"].shape, [4, self.output_length, self.num_mels]
+        )
+        self.assertAllEqual(
+            x["decoder_token_ids"],
+            [[9, 14, 13, 11, 0, 1, 2, 0, 3, 10, 10, 10]] * 4,
+        )
+        self.assertAllEqual(
+            x["decoder_padding_mask"],
+            [[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0]] * 4,
+        )
+
+    def test_preprocess_labeled_batch(self):
+        x = {
+            "encoder_audio": tf.ones((4, 200)),
+            "decoder_text": tf.constant([" airplane at airport"] * 4),
+        }
+        y_in = tf.constant([1] * 4)
+        sw_in = tf.constant([1.0] * 4)
+        x, y, sw = self.preprocessor(x, y_in, sw_in)
+        self.assertAllEqual(
+            x["encoder_features"].shape, [4, self.output_length, self.num_mels]
+        )
+        self.assertAllEqual(
+            x["decoder_token_ids"],
+            [[9, 14, 13, 11, 0, 1, 2, 0, 3, 10, 10, 10]] * 4,
+        )
+        self.assertAllEqual(
+            x["decoder_padding_mask"],
+            [[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0]] * 4,
+        )
+        self.assertAllEqual(y, y_in)
+        self.assertAllEqual(sw, sw_in)
+
+    def test_preprocess_dataset(self):
+        x = {
+            "encoder_audio": tf.ones((4, 200)),
+            "decoder_text": tf.constant([" airplane at airport"] * 4),
+        }
+        ds = tf.data.Dataset.from_tensor_slices(x)
+        ds = ds.map(self.preprocessor)
+        x = ds.batch(4).take(1).get_single_element()
+        self.assertAllEqual(
+            x["encoder_features"].shape, [4, self.output_length, self.num_mels]
+        )
+        self.assertAllEqual(
+            x["decoder_token_ids"],
+            [[9, 14, 13, 11, 0, 1, 2, 0, 3, 10, 10, 10]] * 4,
+        )
+        self.assertAllEqual(
+            x["decoder_padding_mask"],
+            [[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0]] * 4,
+        )
+
+    def test_sequence_length_override(self):
+        input_data = {
+            "encoder_audio": tf.ones((200,)),
+            "decoder_text": " airplane at airport",
+        }
+        x = self.preprocessor(input_data, decoder_sequence_length=6)
+        self.assertAllEqual(x["decoder_token_ids"], [9, 14, 13, 11, 0, 10])
+
+    def test_serialization(self):
+        config = keras.utils.serialize_keras_object(self.preprocessor)
+        new_preprocessor = keras.utils.deserialize_keras_object(config)
+        self.assertEqual(
+            new_preprocessor.get_config(),
+            self.preprocessor.get_config(),
+        )
+
+    @parameterized.named_parameters(
+        ("tf_format", "tf", "model"),
+        ("keras_format", "keras_v3", "model.keras"),
+    )
+    @pytest.mark.large
+    def test_saved_model(self, save_format, filename):
+        input_data = {
+            "encoder_audio": tf.ones((1, 200)),
+            "decoder_text": tf.constant([" airplane at airport"]),
+        }
+        inputs = keras.Input(dtype="string", shape=())
```

Comment:
Check the logs, but I think there is a real error for this. We want inputs to look something like this?

```
        inputs = {
            "encoder_audio": keras.Input(dtype="float32", shape=(??)),
            "decoder_text": keras.Input(dtype="string", shape=(None,)),
        }
```